addons/l10n_it_edi_withholding/models/account_move.py

```diff
@@ -252,3 +239,61 @@ def _l10n_it_edi_import_line(self, element, move_line_form, extra_info=None):
                         move_line_form.tax_ids |= pension_fund_tax
 
         return messages_to_log
+
+    def _l10n_it_edi_import_invoice(self, invoice, data, is_new):
+        res = super()._l10n_it_edi_import_invoice(invoice=invoice, data=data, is_new=is_new)
+        if not res:
+            return
+        self = res
+        tree = data['xml_tree']
+        parent_element = None
+        enasarco_count = 0
+        enasarco_global_line_numero = 0
+        enasarco_amount = 0.0
+        price_unit = 0.0
+        base_amount = 0.0
+        vat_rate = 0
+        # Handle the case where ENASARCO pension fund contribution should be applied on the invoice globally.
+        # In this case, there should only be one element with ENASARCO and these conditions should be fulfilled:
+        # - AliquotaIVA is defined
+        # - PrezzoUnitario == 0.0
+        # - RiferimentoNumero > 0
+        # - a corresponding DatiRiepilogo with the same AliquotaIVA and a ImponibileImporto
+        for other_data_element in tree.xpath('//AltriDatiGestionali'):
+            data_kind_element = other_data_element.xpath('./TipoDato')
+            text_element = other_data_element.xpath('./RiferimentoTesto')
+            if not data_kind_element or not text_element:
+                continue
+            data_kind, data_text = data_kind_element[0].text.lower(), text_element[0].text.lower()
+            if data_kind == 'cassa-prev' and ('enasarco' in data_text or 'tc07' in data_text):
+                enasarco_count += 1
```

Comment:
Use a list of XML elements, so you can have the `enasarco_global_line_numero` and the `enasarco_count` from it.